File: challenges/4-fizz-buzz/elixir/citadin/fizz_buzz.exs
defmodule Challenge do
  def fizz_buzz(num), do: Enum.each(1..num, &verify_multiple/1)

  defp verify_multiple(num) when rem(num, 5) == 0 and rem(num, 3) == 0, do: IO.puts("FizzBuzz")
  defp verify_multiple(num) when rem(num, 5) == 0, do: IO.puts("Fizz")
  defp verify_multiple(num) when rem(num, 3) == 0, do: IO.puts("Buzz")
  defp verify_multiple(num), do: IO.puts(num)
end

File: challenges/4-fizz-buzz/java/maikella/FizzBuzz.java
import java.util.Scanner;

public class FizzBuzz {
   public static void main(String[] args) {
      fizzBuzz();
   }

   private static void fizzBuzz(){

      Scanner s = new Scanner(System.in);

      System.out.println("Entre com um número: ");
      int num = s.nextInt();

      for (int i = 1; i <= num; i++) {

         if(i % 3 == 0 && i % 5 == 0){
            System.out.println("FizzBuzz");
         }else if(i % 3 == 0){
            System.out.println("Fizz");
         }else if(i % 5 == 0){
            System.out.println("Buzz");
         }else {
            System.out.println(i);

         }
      }
   }
}
